api/src/main/java/org/apache/cloudstack/api/command/admin/diagnostics/RunDiagnosticsCmd.java

```diff
@@ -124,14 +126,20 @@ public long getEntityOwnerId() {
 
     @Override
     public String getEventType() {
-        VirtualMachine vm = _entityMgr.findById(VirtualMachine.class, getId());
-        if (vm.getType() == VirtualMachine.Type.ConsoleProxy) {
-            return EventTypes.EVENT_PROXY_DIAGNOSTICS;
-        } else if (vm.getType() == VirtualMachine.Type.SecondaryStorageVm){
-            return EventTypes.EVENT_SSVM_DIAGNOSTICS;
-        } else {
-            return EventTypes.EVENT_ROUTER_DIAGNOSTICS;
+        VirtualMachine.Type vmType = _entityMgr.findById(VirtualMachine.class, getId()).getType();
+        String eventType = "";
+        switch (vmType) {
+            case ConsoleProxy:
+                eventType =  EventTypes.EVENT_PROXY_DIAGNOSTICS;
+            break;
+            case SecondaryStorageVm:
+                eventType = EventTypes.EVENT_SSVM_DIAGNOSTICS;
+                break;
+            case DomainRouter:
```

Comment:
Did you not want this one to be the default, catch-all, as when you had the else statement? There is a 'default' keyword for that.

Reply:
For now, I think it is ok like this so that it is explicit about what is supported, otherwise it might be misleading since it will default to DomainRouter even for User VMs etc.